frontend/src/scenes/surveys/surveyLogic.tsx

```diff
@@ -722,11 +724,19 @@ export const surveyLogic = kea<surveyLogicType>([
             (surveyRatingResults) => {
                 if (surveyRatingResults) {
                     const questionIdx = Object.keys(surveyRatingResults)[0]
-                    const questionResults: number[] = surveyRatingResults[questionIdx].data
-                    if (questionResults.length === 11) {
-                        const promoters = questionResults.slice(9, 11).reduce((a, b) => a + b, 0)
-                        const passives = questionResults.slice(7, 9).reduce((a, b) => a + b, 0)
-                        const detractors = questionResults.slice(0, 7).reduce((a, b) => a + b, 0)
+                    const questionResults = surveyRatingResults[questionIdx]
+
+                    // If we don't have any results, return 'No data available' instead of NaN.
+                    // Context: https://posthog.slack.com/archives/C034XD440RK/p1717193765035719
+                    if (questionResults.total === 0) {
+                        return 'No data available'
+                    }
+
+                    const data: number[] = questionResults.data
+                    if (data.length === 11) {
+                        const promoters = data.slice(9, 11).reduce((a, b) => a + b, 0)
+                        const passives = data.slice(7, 9).reduce((a, b) => a + b, 0)
+                        const detractors = data.slice(0, 7).reduce((a, b) => a + b, 0)
```

Comment:
this fixes a bug I called out [in slack](https://posthog.slack.com/archives/C034XD440RK/p1717193765035719) where we were incorrectly trying to cast `undefined` to a string, which resulted in NaN
![image](https://github.com/PostHog/posthog/assets/4853149/68f48e9c-9290-4e56-ab93-9280fe5a2136)

With my fix, we now get a screen like this
![image](https://github.com/PostHog/posthog/assets/4853149/d3763eb4-7a9e-4094-9f3f-28c2a4144eef)